azkaban-common/src/main/java/azkaban/executor/ExecutableFlow.java

```diff
@@ -408,6 +435,8 @@ public void fillExecutableFromMapObject(
       }
       this.slaOptionStr = slaBuilder.toString();
     }
+    this.userDefinedRetryCount = flowObj.getInt(USER_DEFINED_RETRY_COUNT_PARAM);
+    this.systemDefinedRetryCount = flowObj.getInt(SYSTEM_DEFINED_RETRY_COUNT_PARAM);
```

Comment:
It looks like the default value for integer is -1. Could that lead to allowing an extra retry? Should we set default value to zero?

Reply:
good catch, will update